static/js/hashchange.js

```diff
@@ -312,7 +314,13 @@ function hashchanged(from_reload, e) {
         return undefined;
     }
 
-    if (hash_util.is_overlay_hash(window.location.hash)) {
+    const is_hash_web_public_compatible = hash_util.is_web_public_compatible(current_hash);
+    if (page_params.is_spectator && !is_hash_web_public_compatible) {
+        login_to_access.show();
```

Comment:
Do we want to pass the hash into `login_to_access`, so that it can render a link to `/login/?next=/{hash}`?  That would make links work if you needed to login to use them.  

(Seems fine to address this in a follow-up commit).

Reply:
Ok did it.